### PR TITLE
Addresses the undefined attribute set to the returned WCS object by fit_wcs_from_points function

### LINKED ISSUE #15272: fit_wcs_from_points function sets undefined attribute to the returned WCS object
### Description

The `fit_wcs_from_points` function sets an undefined attribute, `cdelt` to the returned WCS object. `cdelt` attribute is only expected to be found as part of the `Wcsprm` object. 

### Expected behavior

The returned WCS object should not have `cdelt` attribute. 

### How to Reproduce

```python
import numpy as np
import astropy.units as u

from astropy.coordinates import SkyCoord  
from astropy.wcs.utils import fit_wcs_from_points, celestial_frame_to_wcs

xy = (np.array([2810.156, 2810.156,  650.236, 1820.927, 3425.779, 2750.369,
                212.422, 1146.91 ,   27.055, 2100.888,  648.149,   22.212,
                2003.314,  727.098,  248.91 ,  409.998, 1986.931,  128.925,
                1106.654, 1502.67 ]),
      np.array([1670.347, 1670.347,  360.325,  165.663,  900.922,  700.148,
                1416.235, 1372.364,  398.823,  580.316,  317.952,  733.984,
                339.024,  234.29 , 1241.608,  293.545, 1794.522, 1365.706,
                583.135,   25.306]))
ra, dec = (np.array([246.75001315, 246.75001315, 246.72033646, 246.72303144,
                   246.74164072, 246.73540614, 246.73379121, 246.73761455,
                   246.7179495 , 246.73051123, 246.71970072, 246.7228646 ,
                   246.72647213, 246.7188386 , 246.7314031 , 246.71821002,
                   246.74785534, 246.73265223, 246.72579817, 246.71943263]),
         np.array([43.48690547,  43.48690547,  43.46792989,  43.48075238,
                   43.49560501,  43.48903538,  43.46045875,  43.47030776,
                   43.46132376,  43.48252763,  43.46802566,  43.46035331,
                   43.48218262,  43.46908299,  43.46131665,  43.46560591,
                   43.47791234,  43.45973025,  43.47208325,  43.47779988]))
radec = SkyCoord(ra, dec, unit=(u.deg, u.deg))

placeholder_wcs = celestial_frame_to_wcs(frame=radec.frame, projection="TAN")
estimated_wcs = fit_wcs_from_points(xy, radec, projection=placeholder_wcs)

print(estimated_wcs.cdelt)
print(placeholder_wcs.cdelt)
```
``` Python
(1.0, 1.0)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[9], line 5
      2 estimated_wcs = fit_wcs_from_points(xy, radec, projection=placeholder_wcs)
      4 print(estimated_wcs.cdelt)
----> 5 print(placeholder_wcs.cdelt)

AttributeError: 'WCS' object has no attribute 'cdelt'
```

### Versions

```
Linux-6.4.12-200.fc38.x86_64-x86_64-with-glibc2.37
Python 3.9.17 | packaged by conda-forge | (main, Aug 10 2023, 07:02:31) 
[GCC 12.3.0]
astropy 5.3.2
Numpy 1.25.2
pyerfa 2.0.0.3
Scipy 1.11.1
Matplotlib 3.7.1
```

### PR DESCRIPTION
**Pull request by prajwel**
### Description

This pull request is to address https://github.com/astropy/astropy/issues/15272

Fixes #15272

### CODE REVIEW

**Review by mcara — CHANGES_REQUESTED** [review]
Looks good but needs a changelog entry

**Comment by mcara** [comment]
Actually, please wait with the change log. I would like to review more of this code as it does not look right to me.

File: astropy/wcs/utils.py [hunk 1]
```diff
@@ -1092,7 +1092,7 @@ def fit_wcs_from_points(
         wcs = celestial_frame_to_wcs(frame=world_coords.frame, projection=projection)
     else:  # if projection is not string, should be wcs object. use as template.
         wcs = copy.deepcopy(projection)
-        wcs.cdelt = (1.0, 1.0)  # make sure cdelt is 1
+        wcs.wcs.cdelt = (1.0, 1.0)  # make sure cdelt is 1
         wcs.sip = None
 
     # Change PC to CD, since cdelt will be set to 1
```
Comment (prajwel) [inline astropy/wcs/utils.py:1098]: I think the PC-to-CD conversion is done incorrectly. However, the CD values will be used just as an initial guess.

Comment (pllim) [reply astropy/wcs/utils.py:1098]: If this fix is indeed correct, we need a regression test. But let's wait for @mcara .

**Comment by pllim** [comment]
@mcara , did you ever got to verify if the proposed patch is correct? 🙏

**Comment by prajwel** [comment]
Please let me know if any clarification/input is required from my side.

**Comment by pllim** [comment]
A rebase would reset the bot timer while we wait for @mcara .

**Comment by mcara** [comment]
> Looks good but needs a changelog entry

I was waiting for another commit to add change log entry. @pllim, Do we need one here? (I thought we do require for bugs)

**Comment by pllim** [comment]
@mcara , I thought we were waiting for you to decide whether this patch is correct in the first place? See https://github.com/astropy/astropy/pull/15273#issuecomment-1704368862

And yes, if this patch is correct, then it needs a change log.

**Comment by mcara** [comment]
The patch is very correct. Needs a change log entry.

**Comment by pllim** [comment]
@prajwel , are you able to:

1. Add a change log entry for bug fix?
2. Add a test since you already have the use case from https://github.com/astropy/astropy/issues/15272 ?

**Comment by prajwel** [comment]
> @prajwel , are you able to:
> 
>     1. Add a change log entry for bug fix?
> 
>     2. Add a test since you already have the use case from [fit_wcs_from_points function sets undefined attribute to the returned WCS object #15272](https://github.com/astropy/astropy/issues/15272) ?

Ok, I will do it.

**Comment by prajwel** [comment]
I have a change log and a test, as suggested.

File: docs/changes/wcs/15273.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed a bug in ``fit_wcs_from_points`` that sets an undefined attribute to the returned WCS object.
```
Comment (mcara) [inline docs/changes/wcs/15273.bugfix.rst:1]:
```suggestion
Fixed a bug in ``fit_wcs_from_points`` that does not set the default value of the `cdelt` of the returned WCS object.
```
I think this wording is more specific (mentions `cdelt` + it is not clear what an undefined attribute is or why it is bad to set it). My suggestion is just that. Please feel free to adjust the sentence as you wish.

**Review by mcara — APPROVED** [review]
I left a minor suggestion but it looks good otherwise.

File: docs/changes/wcs/15273.bugfix.rst (hunk 2 above)
Comment (prajwel) [reply docs/changes/wcs/15273.bugfix.rst]: Thank you for your comment and I agree. The sentence has been modified as you suggested.

**Comment by mcara** [comment]
`readthedocs` test is failing: as far as I can see it is not related to this PR. However, I have noticed `cdelt`  has a single pair of back quotes. Could you, please, use double back quotes?

File: docs/changes/wcs/15273.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed a bug in ``fit_wcs_from_points`` that does not set the default value of the `cdelt` of the returned WCS object.
```
Comment (mcara) [inline docs/changes/wcs/15273.bugfix.rst:1]:
```suggestion
Fixed a bug in ``fit_wcs_from_points`` that does not set the default value of the ``cdelt`` of the returned WCS object.
```

**Comment by prajwel** [comment]
> `readthedocs` test is failing: as far as I can see it is not related to this PR. However, I have noticed `cdelt` has a single pair of back quotes. Could you, please, use double back quotes?

The readthedocs test still fails. I think the problem could be something else altogether. Maybe @pllim can help here.

**Comment by pllim** [comment]
I haven't seen this failure in `main` nor other PRs. So here are my suggestions:

1. Try to rebase first and see if RTD would be green again. (Not pull, but rebase.)
2. If that does not work, see if `examples/coordinates/plot_obs-planning.py` still would run without error locally for you with this patch.

Thanks!

**Comment by mcara** [comment]
I do not see how the failure can be related to this PR.

**Comment by pllim** [comment]
OK I restarted RTD build. 🤞

**Comment by pllim** [comment]
The coverage stats is a bit too weird. I also restarted the job that do coverage.

**Comment by pllim** [comment]
The RTD failure is too weird. I think this PR needs a rebase.

```
git fetch upstream main
git rebase upstream/main
```

If needed, resolve conflicts. And then you have to force push. If you are worried you will mess this up, make a backup of your branch first. Thanks!

**Comment by prajwel** [comment]
Ok, I messed up!

**Comment by mcara** [comment]
> Ok, I messed up while trying to rebase it. Can I delete this pull request and submit a new one? 😭

Yes.

**Review by mcara — CHANGES_REQUESTED** [review]
Please close this PR and open a new one. Thanks!

**Comment by pllim** [comment]
Sorry that happened! Hopefully you have a backup branch to recover from? Please let me know if you need help!

**Comment by pllim** [comment]
p.s. Usually PR comment log shows you the hashes before and after force push, but I can't seem to easily find it here now. 😱

**Comment by prajwel** [comment]
@pllim 😅 Everything is fine now; I made a different pull request.

**Comment by pllim** [comment]
@prajwel , thank you for your patience and contribution!

**Comment by mcara** [comment]
> Sorry that happened! Hopefully **you have a backup branch** to recover from?

@pllim you are an optimist

**Pull request closed without merging**